### ISSUE TITLE
Quill input / textarea data-placeholder text is not read back for screen readers

### ISSUE THREAD

**Issue by bBaxta** [slab/quill#4495]
When testing the Quill editor with the NVDA screen reader and keyboard, when I set focus to the input / textarea, the screen reader announces:

> section  multi line  editable  blank

There are 2 related issues here.

1. Related to #4434 - a "div" element is used in place of a native input control by using the global attribute "contenteditable=true". This "div" should contain a role="textbox", as this would help identify the purpose of this focusable element with assistive technologies. ![image](https://github.com/user-attachments/assets/0c8f4e57-5615-4069-8132-960a65158072)

2. The data-placeholder text is not being read back. I've addressed this on my own end by adding an aria-labelledby attribute referencing a hidden span containing the text to read out as the placeholder. In my example snippet above, the text that should be read out is "Compose an epic..." per the attribute value.

**Steps for Reproduction**

1. Visit [https://quilljs.com/playground/snow](https://quilljs.com/playground/snow)
2. Start NVDA screen reader.
3. Tab into the Quill editor on the right hand side using your keyboard and set focus to the editable content area.
![image](https://github.com/user-attachments/assets/1f88dc9d-20ff-43e4-aefc-77aecbcd34d7)
4. Note what is read out from the screen reader.

**Expected behavior**:
The place holder text should be read out.

**Actual behavior**:
The place holder text is not read out.

**Platforms**:
Windows 11, Chrome Version 131.0.6778.70 (Official Build) (64-bit)

**Version**:
*, playground is version 2.0.2
